src/client/pmix_client_get.c

```diff
@@ -394,7 +394,7 @@ static void _getnb_cbfunc(struct pmix_peer_t *pr,
             /* fetch the data from my peer module - since we are passing
              * it back to the user, we need a copy of it */
             cb->copy = true;
-            PMIX_GDS_FETCH_KV(rc, pmix_globals.mypeer, cb);
+            PMIX_GDS_FETCH_KV(rc, pmix_client_globals.myserver, cb);
```

Comment:
The comment was changed. Here the server's GDS is used, because the server puts all the requested keys into `dstore` and responds to the client with an empty message (only status, without keys), then the client finds the requested keys in `dstore`. Otherwise, without `dstore`, the server's GDS will be the same as the client.